Contributing.md

```diff
@@ -0,0 +1,31 @@
+Licensing
+=========
+
+The code is distributed under the BSD 2-clause license. Contributors making pull requests must agree that they are able and willing to put their contributions under that license.
+
+Contribution checklist
+======================
+
+* Check the documentation for how to write [a new lexer](https://pygments.org/docs/lexerdevelopment/), [a new formatter](https://pygments.org/docs/formatterdevelopment/) or [a new filter](https://pygments.org/docs/filterdevelopment/)
+* When writing rules, try to merge simple rules. For instance, combine:
+
+  ```python
+  _PUNCTUATION = [
+    (r"\(", token.Punctuation),
+    (r"\)", token.Punctuation),
+    (r"\[", token.Punctuation),
+    (r"\]", token.Punctuation),
+    ("{", token.Punctuation),
+    ("}", token.Punctuation),
+  ]
+  ```
+
+  into:
+
+  ```python
+  (r"[\(\)\[\]{}]", token.Punctuation)
+  ```
+* Be careful with ``.*``. This matches greedily as much as it can. For instance, a rule like ``@.*@`` will match the whole string ``@first@ second @third@``, instead of matching ``@first@`` and ``@second@``. You can use ``@.*?@`` in this case to stop early. The ``?`` tries to match _as few times_ as possible.
+* Don't add your new lexer to ``compiled.py`` -- this file is there for backwards compatibility.
```

Comment:
this can be expanded into "don't add imports of your new lexer anywhere".

Reply:
Rephrased, hope it's clear now.